testutil/testclient/testeventsquery/connection.go

```diff
@@ -47,3 +48,66 @@ func NewOneTimeMockDialer(
 
 	return dialerMock
 }
+
+// NewTwiceMockConnAndDialer returns a new mock connection and mock dialer that
+// will return the mock connection when DialContext is called. The mock dialer
+// will expect DialContext to be called any times. The connection mock will
+// expect Close and Send to be called exactly twice.
+func NewTwiceMockConnAndDialer(
+	t *testing.T,
+	connClosed *atomic.Bool,
+	delayEvent *atomic.Bool,
+) (*mockclient.MockConnection, *mockclient.MockDialer) {
+	connMock := NewTwiceCloseSendConnectionMock(t, connClosed, delayEvent)
+	dialerMock := NewAnyTimesMockDailer(t, connMock)
+	return connMock, dialerMock
+}
+
+// NewTwiceCloseSendConnectionMock returns a mock connection that will expect
+// Close and Send to be called exactly twice. The connection mock will set the
+// connClosed atomic to true when Close is called and false when Send is called.
+// The connection mock will set the delayEvent atomic to false when Send is
+// called. This is to allow the caller to subscribe to the first event emitted
+func NewTwiceCloseSendConnectionMock(
+	t *testing.T,
+	connClosed *atomic.Bool,
+	delayEvent *atomic.Bool,
+) *mockclient.MockConnection {
+	ctrl := gomock.NewController(t)
+	connMock := mockclient.NewMockConnection(ctrl)
+	// Expect the connection to be closed and the dialer to be re-established
+	connMock.EXPECT().Close().DoAndReturn(func() error {
+		connClosed.CompareAndSwap(false, true)
+		return nil
+	}).
+		// Close once by hand and the EventQueryClient will close it again
+		Times(2)
```

Comment:
NIT: Fix spacing

Reply:
This is the correct spacing from gofumpt?